src/hdr-config.cpp

```diff
@@ -0,0 +1,521 @@
+
+            uint32_t exposure_value = static_cast<uint32_t>(_hdr_sequence_params[i]._exposure);
+            frames_config.insert(frames_config.end(), &CONTROL_ID_EXPOSURE, &CONTROL_ID_EXPOSURE + 1);
+            frames_config.insert(frames_config.end(), (uint8_t*)&exposure_value, (uint8_t*)&exposure_value + 4);
+            
+            uint32_t gain_value = static_cast<uint32_t>(_hdr_sequence_params[i]._gain);
+            frames_config.insert(frames_config.end(), &CONTROL_ID_GAIN, &CONTROL_ID_GAIN + 1);
+            frames_config.insert(frames_config.end(), (uint8_t*)&gain_value, (uint8_t*)&gain_value + 4);
+        }
+
+        return frames_config;
+    }
+
+    bool hdr_config::validate_config() const
+    {
+        // to be elaborated or deleted
+        return true;
+    }
+
+    void hdr_config::set_id(float value)
+    {
+        int new_id = static_cast<int>(value);
+        
+        if (new_id != _id)
+        {
+            _id = new_id;
+        }
+    }
+
+    void hdr_config::set_sequence_size(float value)
+    {
+        size_t new_size = static_cast<size_t>(value);
+        if (new_size > 3 || new_size < 2)
+            throw invalid_value_exception(to_string() << "hdr_config::set_sequence_size(...) failed! Only size 2 or 3 are supported.");
+
+        if (new_size != _sequence_size)
+        {
+            _hdr_sequence_params.resize(new_size);
+            _sequence_size = new_size;
+        }       
+    }
+
+    void hdr_config::set_sequence_index(float value)
+    {
+        size_t new_index = static_cast<size_t>(value);
+        
+        _is_config_in_process = (new_index != 0);
+
+        if (new_index <= _hdr_sequence_params.size())
+        {
+            _current_hdr_sequence_index = new_index - 1;
+            _has_config_changed = true;
+        }
+        else
+            throw invalid_value_exception(to_string() << "hdr_config::set_sequence_index(...) failed! Index above sequence size.");
+    }
+
+    void hdr_config::set_exposure(float value)
+    {
+        /* TODO - add limitation on max exposure to be below frame interval - is range really needed for this?*/
+        _hdr_sequence_params[_current_hdr_sequence_index]._exposure = value;
+        _has_config_changed = true;
+    }
+
+    void hdr_config::set_gain(float value)
+    {
+        _hdr_sequence_params[_current_hdr_sequence_index]._gain = value;
+        _has_config_changed = true;
+    }
+
+
+    hdr_params::hdr_params() :
+        _sequence_id(0),
+        _exposure(0.f),
+        _gain(0.f)
+    {}
+    
+    hdr_params::hdr_params(int sequence_id, float exposure, float gain) :
+        _sequence_id(sequence_id),
+        _exposure(exposure),
+        _gain(gain)
+    {}
+
+    hdr_params& hdr_params::operator=(const hdr_params& other)
+    {
+        _sequence_id = other._sequence_id;
+        _exposure = other._exposure;
+        _gain = other._gain;
+
+        return *this;
+    }
+
+    
+
+    // explanation for the sub-preset:
```

Comment:
Why? It can really help to understand, I think.
A better way may be to add a class which has the same structure as the firmware's. But we'd  better not delete the comment without this I think.

Reply:
Accepted